cqf-fhir-cr-hapi/src/main/java/org/opencds/cqf/fhir/cr/hapi/cdshooks/discovery/CrDiscoveryService.java

```diff
@@ -81,7 +82,18 @@ public ILibraryAdapter resolvePrimaryLibrary(IPlanDefinitionAdapter planDefiniti
                     fhirVersion(), planDefinition.getLibrary().get(0));
         }
         if (canonical != null) {
-            return adapterFactory.createLibrary(SearchHelper.searchRepositoryByCanonical(repository, canonical));
+            IBaseResource library;
```

Comment:
Nitpick:  declare this as final to indicate you expect the if/else block to exhaust all possible initialization code paths.